### ISSUE TITLE
quoteToOrder converter missing productId because of CartItemExtension

### ISSUE THREAD

**Issue by experius-nl** [SpotlerSoftware/magento2-mailplus-connector#12]
Because of the CartItem extension attribute product_id in this module, the quoteToOrder converter is missing the productId. This leads to potential errors, which we found out by combining with Anowave/Ec module. We found the order items to no longer have productId's. We've managed to track this to this di:
`<extension_attributes for="Magento\Quote\Api\Data\CartItemInterface">
        <attribute code="product_id" type="string">
        </attribute>
    </extension_attributes>`

This only happens in Magento 2.3.3, this does not happen in Magento 2.3.2.

**Comment by experius-nl** [comment]
Related file is here: https://github.com/magento/magento2/blob/2.3.3/lib/internal/Magento/Framework/DataObject/Copy.php

**Comment by 808brinks** [comment]
Hi @experius-nl,
Could you describe how the extension attribute on CartItems is causing this problem?
Are both plugins defining the same extension attribute or is there some other problem?

**Comment by experius-nl** [comment]
During checkout, the quote is being converted to an order in the database. During this process, for each product the product_id is retrieved and stored in sales_order_item. Because of the extension attribute being named product_id, which already exists in Magento as a regular attribute, Magento now tries to look for a non-existent extension attribute called product_id, instead of the regular product_id. Of course, this results to null.

At this point, our Anowave module tries to load the product as well, using the product_id. Because every product in the order is supposed to always have one, there's no safety checking. Thus, it tries to use null as an argument where a product is required.

This is how we found out, but you can just build an observer module that triggers on order_save, and have it use the sales_order_item model to display the productnames of the order. That'll crash too.

**Comment by 808brinks** [comment]
I'm not quite convinced that it's our product_id extension attribute that is conflicting with the standard magento checkout process.

As far as i can see the cartItem doesn't have a product_id field. ([docs](https://github.com/pepe1518/magento2/blob/master/vendor/magento/module-quote/Api/Data/CartItemInterface.php)). Which is why our plugin adds it in the first place.

Also I don't think (not sure) that extension attributes overwrite or add properties from the original class. They should be added to the extension_attributes property. See example

```
                 {
                    "item_id": 5,
                    "sku": "MS08-XS-Black",
                    "qty": 1,
                    "name": "Strike Endurance Tee",
                    "price": 39,
                    "product_type": "configurable",
                    "quote_id": "3",
                    "extension_attributes": {
                        "product_id": "628"
                    }
                }
```
I'm not a magento developer so I might be wrong about all this. Please let me know if i am.

**Comment by experius-nl** [comment]
![Untitled](https://user-images.githubusercontent.com/5173161/69240026-e222f680-0b9b-11ea-8bc8-941cb320344d.png)

This is a screenshot of the sales_order_item table. We did 2 attempts at an order, which are the first two rows of the image. Item_id 133737 has a product_id of null (which leads to errors), and item_id 133738 has a product_id of 214 (the correct id).

The only difference between the two attempts is that we commented the following parts of your mailplus module, in extension_attributes.xml and di.xml in between the attempts:

``
``

Disabling this functionality fixes the bug.
